uuid.go

```diff
@@ -62,3 +62,58 @@ func (g *Generator) Next() [24]byte {
 	binary.LittleEndian.PutUint64(uuid[:8], x)
 	return uuid
 }
+
+// Hex128 is a convenience method that returns Hex128(g.Next()).
+func (g *Generator) Hex128() string {
+	return Hex128(g.Next())
+}
+
+// Hex128 returns an RFC4122-like representation of the
+// first 128 bits of the given UUID. For example:
+//
+//	f81d4fae-7dec-11d0-a765-00a0c91e6bf6.
+//
+// It does not bother to set the version or variant bits
+// of the UUID, as they only serve to reduce the randomness.
+//
+// If you want unpredictable UUIDs, you might want to consider
+// hashing the uuid (using SHA256, for example) before passing it
+// to Hex128.
+func Hex128(uuid [24]byte) string {
+	b := make([]byte, 36)
+	hex.Encode(b[0:8], uuid[0:4])
+	b[8] = '-'
+	hex.Encode(b[9:13], uuid[4:6])
+	b[13] = '-'
+	hex.Encode(b[14:18], uuid[6:8])
+	b[18] = '-'
+	hex.Encode(b[19:23], uuid[8:10])
+	b[23] = '-'
+	hex.Encode(b[24:], uuid[10:16])
+	return string(b)
+}
+
+// ValidHex128 reports whether id is a valid UUID as returned by Hex128.
+func ValidHex128(id string) bool {
+	if len(id) != 36 {
+		return false
+	}
+	if id[8] != '-' || id[13] != '-' || id[18] != '-' || id[23] != '-' {
+		return false
+	}
+	return isValidHex(id[0:8]) &&
+		isValidHex(id[9:13]) &&
+		isValidHex(id[14:18]) &&
+		isValidHex(id[19:23]) &&
+		isValidHex(id[24:])
+}
+
+func isValidHex(s string) bool {
+	for i := 0; i < len(s); i++ {
+		c := s[i]
+		if !('0' <= c && c <= '9' || 'a' <= c && c <= 'f') {
```

Comment:
Upper case isn't valid then?

Reply:
Yeah, I didn't think it was needed as none of the standard UUID packages produce upper case. I adjusted the doc comment to make that clear.